internal/storage/v2/grpc/handler.go

```diff
@@ -21,13 +20,11 @@ import (
 var (
 	_ storage.TraceReaderServer      = (*Handler)(nil)
 	_ storage.DependencyReaderServer = (*Handler)(nil)
-	_ ptraceotlp.GRPCServer          = (*Handler)(nil)
 )
 
 type Handler struct {
 	storage.UnimplementedTraceReaderServer
 	storage.UnimplementedDependencyReaderServer
-	ptraceotlp.UnimplementedGRPCServer
 
 	traceReader tracestore.Reader
 	traceWriter tracestore.Writer
```

Comment:
do we need the writer?

Reply:
what happens with archive storage, does it need the writer?

Reply:
> what happens with archive storage, does it need the writer?
Is there a reason the archive storage would need it? After we did the refactor, there is no difference in archive and regular storage

Reply:
Archive storage is used to save a trace from the UI. If we're testing query service config with gRPC storage then the archive storage in the `query` binary must be able to perform a write, which it will do by calling the Write API on port `:17272`. So yeah that single gRPC server should be service both read and write endpoints. I don't think we want to have yet another binary with OTLP receiver for that.

NB I don't think the e2e test is actually exercising the "archive trace" workflow, it would be good to add that.